### DISCUSSION TITLE
I need help ! FODs related

### DISCUSSION THREAD

**Discussion by imaneelmissaoui** [dipy/dipy#2809] [3. Community Lounge]
Hello all,
After i constructed the odfs , i want to slice them into convex shapes . After i checked the methods i can apply on fodf_spheres i have this : ['AddConsumer',
 'AddEstimatedRenderTime',
 'AddObserver',
 'AddOrientation',
 'AddPosition',
 'ApplyProperties',
 'B',
 'BreakOnError',
 'BuildPaths',
 'ComputeMatrix',
 'CoordinateSystems',
 'DEVICE',
 'DebugOff',
 'DebugOn',
 'DragableOff',
 'DragableOn',
 'FastDelete',
 'ForceOpaqueOff',
 'ForceOpaqueOn',
 'ForceTranslucentOff',
 'ForceTranslucentOn',
 'GeneralTextureTransform',
 'GeneralTextureUnit',
 'GetActors',
 'GetActors2D',
 'GetAddressAsString',
 'GetAllocatedRenderTime',
 'GetBackfaceProperty',
 'GetBounds',
 'GetCenter',
 'GetClassName',
 'GetCommand',
 'GetConsumer',
 'GetCoordinateSystem',
 'GetCoordinateSystemAsString',
 'GetCoordinateSystemDevice',
 'GetCoordinateSystemRenderer',
 'GetDebug',
 'GetDragable',
 'GetEstimatedRenderTime',
 'GetForceOpaque',
 'GetForceTranslucent',
 'GetGlobalWarningDisplay',
 'GetIsIdentity',
 'GetIsInMemkind',
 'GetLength',
 'GetMTime',
 'GetMapper',
 'GetMatrix',
 'GetModelToWorldMatrix',
 'GetNextPath',
 'GetNumberOfConsumers',
 'GetNumberOfGenerationsFromBase',
 'GetNumberOfGenerationsFromBaseType',
 'GetNumberOfPaths',
 'GetObjectDescription',
 'GetObjectName',
 'GetOrientation',
 'GetOrientationWXYZ',
 'GetOrigin',
 'GetPickable',
 'GetPosition',
 'GetProperty',
 'GetPropertyKeys',
 'GetRedrawMTime',
 'GetReferenceCount',
 'GetRenderTimeMultiplier',
 'GetScale',
 'GetShaderProperty',
 'GetSupportsSelection',
 'GetTexture',
 'GetUseBounds',
 'GetUserMatrix',
 'GetUserTransform',
 'GetUserTransformMatrixMTime',
 'GetUsingMemkind',
 'GetVisibility',
 'GetVolumes',
 'GetXRange',
 'GetYRange',
 'GetZRange',
 'GlobalWarningDisplayOff',
 'GlobalWarningDisplayOn',
 'HasKeys',
 'HasObserver',
 'HasOpaqueGeometry',
 'HasTranslucentPolygonalGeometry',
 'InitPathTraversal',
 'InitializeObjectBase',
 'InvokeEvent',
 'IsA',
 'IsConsumer',
 'IsRenderingTranslucentPolygonalGeometry',
 'IsTypeOf',
 'MakeProperty',
 'Modified',
 'NewInstance',
 'PHYSICAL',
 'Pick',
 'PickableOff',
 'PickableOn',
 'PokeMatrix',
 'ProcessSelectorPixelBuffers',
 'Register',
 'ReleaseGraphicsResources',
 'RemoveAllObservers',
 'RemoveConsumer',
 'RemoveObserver',
 'RemoveObservers',
 'Render',
 'RenderFilteredOpaqueGeometry',
 'RenderFilteredOverlay',
 'RenderFilteredTranslucentPolygonalGeometry',
 'RenderFilteredVolumetricGeometry',
 'RenderOpaqueGeometry',
 'RenderOverlay',
 'RenderTranslucentPolygonalGeometry',
 'RenderVolumetricGeometry',
 'RestoreEstimatedRenderTime',
 'RotateWXYZ',
 'RotateX',
 'RotateY',
 'RotateZ',
 'SafeDownCast',
 'SetAllocatedRenderTime',
 'SetBackfaceProperty',
 'SetCoordinateSystem',
 'SetCoordinateSystemDevice',
 'SetCoordinateSystemRenderer',
 'SetCoordinateSystemToDevice',
 'SetCoordinateSystemToPhysical',
 'SetCoordinateSystemToWorld',
 'SetDebug',
 'SetDragable',
 'SetEstimatedRenderTime',
 'SetForceOpaque',
 'SetForceTranslucent',
 'SetGlobalWarningDisplay',
 'SetIsRenderingTranslucentPolygonalGeometry',
 'SetMapper',
 'SetMemkindDirectory',
 'SetObjectName',
 'SetOrientation',
 'SetOrigin',
 'SetPickable',
 'SetPosition',
 'SetPropertiesFromModelToWorldMatrix',
 'SetProperty',
 'SetPropertyKeys',
 'SetReferenceCount',
 'SetRenderTimeMultiplier',
 'SetScale',
 'SetShaderProperty',
 'SetTexture',
 'SetUseBounds',
 'SetUserMatrix',
 'SetUserTransform',
 'SetVisibility',
 'ShallowCopy',
 'UnRegister',
 'UseBoundsOff',
 'UseBoundsOn',
 'UsesGarbageCollector',
 'VisibilityOff',
 'VisibilityOn',
 'WORLD',
 '__class__',
 '__delattr__',
 '__dict__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__le__',
 '__lt__',
 '__module__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 '__this__',
 '__vtkname__',
 '_generate_color_for_vertices',
 '_get_all_faces',
 '_get_all_vertices',
 '_get_odf_offsets',
 '_get_sf',
 '_get_sphere_directions',
 '_update_mapper',
 'affine',
 'colormap',
 'display',
 'display_extent',
 'faces',
 'global_cm',
 'grid_shape',
 'indices',
 'mapper',
 'mask',
 'odfs',
 'override',
 'radial_scale',
 'set_opacity',
 'slice_along_axis',
 'update_sphere',
 'vertices']
Maybe i can use the slice_along_axis ? i don't see how to do it 
Any advice or help would be much appreciated 

Many thanks,
Imy

**Comment by skoudoro** [comment]
Not sure to understand your question but @guaje can help for this when he find time

**Reply by imaneelmissaoui** [reply]
Thanks for you reply ! Hoping @guaje can help me with it :)) 
To be more clear now i have used csd model to constructs odfs : i have 
csd_odf = csd_fit.odf(default_sphere) and 
fodf_spheres = actor.odf_slicer(csd_odf, sphere=default_sphere, scale=0.9, norm=False)
 
I am trying to divide the odfs into convex shapes to be able to use GJK-EPA algrorithm for polytope expansion and i don't see how i can do it?                              
Also I guess i need to use fodf_spheres not csd_odf ?

Many thanks,
Imy

**Reply by skoudoro** [reply]
`fodf_spheres` provides you all need to build yourself your convex shapes with the `vertices` and `faces` (triangles).

it depends of what is the input of your GJK-EPA algorithm. Not sure what is expected but you have everything with `vertices`,  `faces` and `indices`  which describe the convex shapes of all odf

I hope it helps

**Reply by imaneelmissaoui** [reply]
Thanks for your help! I've been wondering how can i get the center of each odf from fodf_spheres?
